r/R/dplyr-datetime-helpers.R

```diff
@@ -105,7 +105,12 @@ binding_as_date <- function(x,
                             tryFormats = "%Y-%m-%d",
                             origin = "1970-01-01") {
   if (is.null(format) && length(tryFormats) > 1) {
-    abort("`as.Date()` with multiple `tryFormats` is not supported in Arrow")
+    abort(
+      paste(
+        "`as.Date()` with multiple `tryFormats` is not supported in Arrow,",
+        "consider using the specialised parsing functions"
```

Comment:
As a naive user I might not know what you mean here by "the specialised parsing functions" - mind making this message more specific?

Reply:
Does "lubridate specialised parsing functions" sound better?

Reply:
Or the "specialised parsing functions in lubridate"? Maybe with an "(e.g ymd())".

Reply:
I'd be inclined to include an example function (do you mean like `ymd()` etc?) as it's not immediately apparent what functions you mean without a little extra thought.

Reply:
I have updated the error message to: 
```r
"`as.Date()` with multiple `tryFormats` is not supported in Arrow,",
         "consider using the specialised parsing functions in lubridate, such as, `ymd()`"
```

Reply:
I think the message might also need not to specify `as.Date()` as it is raised in the body of a function we use from both the bindings for `as.Date` and `lubridate::as_date()`.

Reply:
Also, sorry if this sounds super-pendantic, but clarity in error messages is important; "in lubridate" implies that we're directing them to do it outside of Arrow rather than being possible within an Arrow context.  Something like "consider using the lubridate specialised parsing functions such as `ymd()`, `ydm()`, etc" might remove this implication.

Reply:
Done. Thanks for the input / review!

Reply:
Marking this as "unresolved" for now as I think you missed my second to last comment there.

Reply:
Sounds good. Done.